Sources/NIOTLS/NIOTypedApplicationProtocolNegotiationHandler.swift

```diff
@@ -0,0 +1,141 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(AsyncChannel) import NIOCore
+import DequeModule
+
+
+@_spi(AsyncChannel)
+public final class NIOTypedApplicationProtocolNegotiationHandler<NegotiationResult>: ChannelInboundHandler, RemovableChannelHandler, NIOProtocolNegotiationHandler {
```

Comment:
We should probably add some docs for this type and the public properties below.

Reply:
Added comments to the type. The property is already documented in the protocol so no need to document it here as well.